elide-core/src/main/java/com/yahoo/elide/core/datastore/inmemory/InMemoryTransaction.java

```diff
@@ -136,19 +137,21 @@ public Object loadObject(Class<?> entityClass, Serializable id,
     public Iterable<Object> loadObjects(Class<?> entityClass, Optional<FilterExpression> filterExpression,
                                         Optional<Sorting> sorting, Optional<Pagination> pagination,
                                         RequestScope scope) {
-        ConcurrentHashMap<String, Object> data = dataStore.get(entityClass);
-
-        // Support for filtering
-        if (filterExpression.isPresent()) {
-            Predicate predicate = filterExpression.get().accept(new InMemoryFilterVisitor(scope));
-            return (Collection<Object>) data.values().stream()
-                    .filter(predicate::test)
-                    .collect(Collectors.toList());
-        }
+        Map<String, Object> data = dataStore.get(entityClass);
+        synchronized (data) {
```

Comment:
We need this to be synchronized every where it's used (i.e. `loadObject` and `commit`). Would it be easier to wrap `LinkedHashMap` and synchronize the entire structure? Or do we want to do the bookkeeping here?